File: src/main/java/com/andrew121410/mc/ccminecraftbot/packets/handle/inventory/OnServerCloseWindowPacket.java
package com.andrew121410.mc.ccminecraftbot.packets.handle.inventory;

import com.andrew121410.mc.ccminecraftbot.CCBotMinecraft;
import com.andrew121410.mc.ccminecraftbot.packets.PacketHandler;
import com.github.steveice10.mc.protocol.packet.ingame.server.window.ServerCloseWindowPacket;
import lombok.AllArgsConstructor;

@AllArgsConstructor
public class OnServerCloseWindowPacket extends PacketHandler<ServerCloseWindowPacket> {

    private CCBotMinecraft CCBotMinecraft;

    @Override
    public void handle(ServerCloseWindowPacket packet) {
        this.CCBotMinecraft.getPlayer().getPlayerInventory().closeWindow(packet.getWindowId());
    }
}

File: src/main/kotlin/com/andrew121410/mc/ccminecraftbot/packets/handle/inventory/OnServerCloseWindowPacket.kt
package com.andrew121410.mc.ccminecraftbot.packets.handle.inventory

import com.andrew121410.mc.ccminecraftbot.CCBotMinecraft
import com.andrew121410.mc.ccminecraftbot.packets.PacketHandler
import com.github.steveice10.mc.protocol.packet.ingame.server.window.ServerCloseWindowPacket

class OnServerCloseWindowPacket : PacketHandler<ServerCloseWindowPacket>() {
    override fun handle(packet: ServerCloseWindowPacket, ccBotMinecraft: CCBotMinecraft) {
        ccBotMinecraft.player.playerInventory.closeWindow(packet.windowId)
    }
}